pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -0,0 +1,278 @@
+			Zone:        opt.Zone,
+		}
+
+		_, err = gcpclient.Disks().Insert(opt.ProjectID, opt.Zone, disk).Context(ctx).Do()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create disk")
+		}
+	}
+
+	logger.Info("Disk created", "disk", opt.DiskName)
+
+	disks := []*compute.AttachedDisk{
+		{
+			AutoDelete: true,
+			Boot:       true,
+			DiskSizeGb: 10,
+			Source:     "projects/" + opt.ProjectID + "/zones/" + opt.Zone + "/disks/" + opt.DiskName,
+			Mode:       "READ_WRITE",
+		},
+	}
+
+	networkInterfaces := []*compute.NetworkInterface{
+		{
+			Network:       "projects/" + opt.ProjectID + "/global/networks/" + opt.Shoot.Name,
+			Subnetwork:    "regions/" + opt.Region + "/subnetworks/" + opt.Subnetwork,
+			AccessConfigs: []*compute.AccessConfig{{Name: "External NAT", Type: "ONE_TO_ONE_NAT"}},
+		},
+	}
+
+	machineType := "zones/" + opt.Zone + "/machineTypes/n1-standard-1"
+
+	instance, err := getBastionInstance(ctx, gcpclient, opt)
+	if err != nil {
+		return nil, err
+	}
+
+	metadataItems := []*compute.MetadataItems{
+		{
+			Key:   "startup-script",
+			Value: pointer.StringPtr(string(bastion.Spec.UserData)),
+		},
+	}
+
+	if instance != nil {
+		logger.Info("Existing bastion compute instance found", "compute_instance_name", instance.Name)
+	} else {
+		instance := &compute.Instance{
+			Disks:              disks,
+			DeletionProtection: false,
+			Description:        "Bastion Instance",
+			Name:               opt.BastionInstanceName,
+			Zone:               opt.Zone,
+			MachineType:        machineType,
+			NetworkInterfaces:  networkInterfaces,
+			Tags:               &compute.Tags{Items: []string{opt.BastionInstanceName}},
+			Metadata:           &compute.Metadata{Items: metadataItems},
+		}
+
+		_, err = gcpclient.Instances().Insert(opt.ProjectID, opt.Zone, instance).Context(ctx).Do()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create instance")
+		}
+	}
+
+	return getInstanceEndpoints(ctx, gcpclient, opt)
+}
+
+func getInstanceEndpoints(ctx context.Context, gcpclient gcpclient.Interface, opt *Options) (*bastionEndpoints, error) {
+	instance, err := getBastionInstance(ctx, gcpclient, opt)
+	if err != nil {
+		return nil, err
+	}
+
+	if instance == nil {
+		return nil, nil
+	}
+
+	if instance.Status != "RUNNING" {
+		err := errors.New("Instance not running, Status:" + instance.Status)
+		return nil, err
+
+	}
+
+	endpoints := &bastionEndpoints{}
+
+	if ingress := addressToIngress(&instance.Name, &instance.NetworkInterfaces[0].NetworkIP); ingress != nil {
+		endpoints.private = ingress
+	}
+
+	if instance.NetworkInterfaces == nil || len(instance.NetworkInterfaces) == 0 {
+		err := errors.New(instance.Name + ":" + "no network interfaces found")
+		return nil, err
+	}
+
+	if instance.NetworkInterfaces[0].AccessConfigs == nil || len(instance.NetworkInterfaces[0].AccessConfigs) == 0 {
+		err := errors.New(instance.Name + ":" + "no access config found for network interface")
```

Comment:
use `fmt.Errorf` when having variable string

Reply:
committed in 362db688cb6fcae409c07490f2b82a6734f14d73